### ISSUE TITLE
No connection with FileZilla in Access Point mode

### ISSUE THREAD

**Issue by anbtzdd** [robert-hh/FTP-Server-for-ESP8266-ESP32-and-PYBD#23]
Hi, it is a very helpfully and easily to use program to provide a small ftp server. Many, many thanks for it.
The server works very fine if I have a wifi client connection on my ESP32. The connection by FileZilla works very fin in this case.
But when I use the ftp-server by access point mode, I get the message "FTP-Server gestartet auf 0.0.0.0:21" an the connection by FilZilla ended with error "Verbindungsversuch fehlgeschlagen mit "ECONNREFUSED - Verbindung durch Server verweigert".". 
The access point has the IP 192.168.4.1

With the command line tool ftp on a Windows PC the connection works fine right away.
Have everyone a little tip, what is the different and was can I do?

**Comment by robert-hh** [comment]
To understand you right: If the board works in AP mode, you cannot connect FTP using Filezilla, but you can using Windows FTP. 
Windows FTP uses active connection mode, while that can be configured in Filezilla with passive mode being preferred.

**Comment by anbtzdd** [comment]
Many thanks for the very fast support.
Yes, after changing to the active mode the connection works also with FileZilla.
Only for my understanding. Why I get the message 'IP 0.0.0.0:21' when I start the FTP server in AP mode and the right IP address in the client mode?

**Comment by robert-hh** [comment]
Actually I do not know right now.

**Comment by anbtzdd** [comment]
Hi Robert,

I have investigated the problem a little.
Therefore, I have inserted a new print-line in line 469. They show the interface, the wlan.activ-status and the wlan.ifconfig-information. 

for interface in [network.AP_IF, network.STA_IF]:
    wlan = network.WLAN(interface)
    print(interface, wlan.active(), wlan.ifconfig())
    if not wlan.active():
       continue

The result is very interesting for me.
1. After wlan = network.WLAN(network.AP_IF) the object wlan contains already at this time all network information like IP-address, etc.
So there are two lines enough to get the follow result (No network was started):

import uftpd
uftpd.start()

1 False ('192.168.4.1', '255.255.255.0', '192.168.4.1', '0.0.0.0')
0 True ('0.0.0.0', '0.0.0.0', '0.0.0.0', '0.0.0.0')

Very interesting is the wlan.active-status of the client-interface 0!?

2. At the second try, the wlan connected before the start of the FTP-Server.
Both interfaces contain IP-addresses, but only the client interface is activated. 

Client Mode:

1 False ('192.168.4.1', '255.255.255.0', '192.168.4.1', '0.0.0.0')
0 True ('192.168.178.63', '255.255.255.0', '192.168.178.1', '192.168.178.234')
FTP-Server gestartet IP: 192.168.178.63, Port: 21

3. At the third try, an access point was started before start the FTP-Server.
Only the AP-interface contain IP-addresses, but both interfaces are active. The result is the wrong message.

Access Point Mode:

1 True ('192.168.4.1', '255.255.255.0', '192.168.4.1', '0.0.0.0')
0 True ('0.0.0.0', '0.0.0.0', '0.0.0.0', '0.0.0.0')
FTP-Server gestartet IP: 0.0.0.0, Port: 21

As it seems, the command "if not wlan.active():" isn't really suitable to check, which interface is active because the status is always True. 
I replaced the for loop with the follow code, and now, it works fine every time.

ap = network.WLAN(network.AP_IF)
wl = network.WLAN(network.STA_IF)
active_network = False

if ap.active():
    active_network = True
    wlan = network.WLAN(network.AP_IF)
elif wl.active():
    if wl.isconnected():
        active_network = True
        wlan = network.WLAN(network.STA_IF) 
            
if active_network:
	ifconfig = wlan.ifconfig()
   addr = socket.getaddrinfo(ifconfig[0], port)
   . . .
